File: services/front/lib/features/profile/services/profile_service.dart
import 'package:front/src/grpc/generated/profile.pb.dart';
import 'package:logger/logger.dart';

import '../clients/profile_client.dart';

final logger = Logger();

class ProfileService {
  final ProfileClient profileClient;

  ProfileService(this.profileClient);

  Future<ProfileObject?> getMy() async {
    return await profileClient.getMyProfile();
  }

  Future<ProfileObject?> getProfile(String profileId) async {
    return await profileClient.getProfile(profileId);
  }
}

File: services/streaming/src/services/profile.service.ts
import * as ProfileClient from "../grpc/clients/profile.client";
import * as profileGrpc from "../grpc/generated/profile";

export const health = async () =>
  await ProfileClient.health();

export const status = async () =>
  await ProfileClient.status();

export const livez = async () =>
  await ProfileClient.livez();

export const readyz = async () =>
  await ProfileClient.readyz();

export const getProfileByUser = async (userId: string): Promise<profileGrpc.ProfileObject | null> =>
  await ProfileClient.getProfileByUser(userId);